src/components/preview/preview.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+* A Preview widget.
+*
+* == How to use a Preview in a component:
+*
+* In your file:
+*
+*   import Preview from 'carbon/lib/components/preview'
+*
+* To render the Preview:
+*
+*   <Preview>
+*     My text content
+*   </Preview>
+*
+*
+*
+* @class Text
+* @constructor
+*/
+class Preview extends React.Component {
+  static propTypes = {
+    /**
+     * Custom className
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * Children elements
+     *
+     * @property children
+     * @type {Node}
+     */
+    children: PropTypes.node,
+
+    /**
+     * The id attribute to use for the text
+     *
+     * @property id
+     * @type {String}
+     */
+    id: PropTypes.string,
+
+    /**
+     * The isVisible attribute to use for the text
+     *
+     * @property id
+     * @type {String}
+     */
+    isVisible: PropTypes.bool
+  };
+
+  static defaultProps = {
+    className: '',
+    isVisible: true
+  };
+
+  /**
+   * Main classes
+   *
+   * @method mainClasses
+   * @return {String} classNames for text
+   */
+  get mainClasses() {
+    return classNames(
+      'carbon-preview',
+      { 'carbon-preview--placeholder': !this.props.children },
```

Comment:
would be cool to have an additional prop to give the developer manual control in case `children` isn't that flag they want it to be based on.

something like

```js
isReady = () => {
  if (typeof this.props.ready !== 'undefined') {
    return this.props.ready;
  }

  return !!this.props.children;
}
```

not sure what the prop should be called... `ready`, `loading`, `preview` .... ?